src/Arbus.Network/Exceptions/NetworkException.cs

```diff
@@ -7,16 +8,24 @@ public class NetworkException : Exception
     public HttpStatusCode? StatusCode { get; }
 
     public ProblemDetails? ProblemDetails { get; }
+    
+    public ApiEndpoint? Endpoint { get; }
```

Comment:
What are the pros and cons of making Endpoint nullable?

Reply:
+:
- In case when you throw NetworkException without endpoint (not in ApiEndpoint or HttpContext) you can just pass null. For this case we need to make ApiEndpoint nullable in constructor too

-:
- If you want to get some info about endpoint (e.g. log endpoint type) you need to check it for null

Reply:
Why would someone throw a NetworkException without interacting with the network?

Reply:
Seems like there's always will be an Endpoint when the NetworkException is thrown and we can make it simpler for users making NetworkException.Endpoint non nullable

Reply:
Hmm, yes. It seems strange. Use this lib and throw NetworkException not in endpoint or HttpContext. This case is available so I took it into account but maybe it redundant

Reply:
I think we also should make constroctor NetworkException(string message) protected. Otherwise user will have a possibility to create exception without endpoint

Reply:
What about situation with INativeHttpClient? It works with HttpRequest(Response)Message and doesn't know anything about ApiEndpoint but it's implementation can throw NetworkExceptions. If we add ApiEndpoint to SendRequest method signature then user will not understand what to use to execute request to endpoint: IHttpContext.RunEndpoint(ApiEndpoint) or INativeHttpClient.Send(ApiEndpoint, Uri, timeout).
So, maybe Endpoint property should stay nullable?